torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -235,45 +270,62 @@ def get_aten_name(
+        overload: Optional[str],
         diagnostic_context: diagnostics.DiagnosticContext,
-    ) -> Set[Union["onnxscript.OnnxFunction", "onnxscript.TracedOnnxFunction"]]:
+    ) -> Tuple[
+        Set[registration.SymbolicFunction], Optional[Set[registration.SymbolicFunction]]
+    ]:
         """Get the function overloads from the registry.
 
         Args:
             node: The node to get the function overloads for.
-            aten_name: The aten name of the node.
+            domain: The domain of the node.
+            op_name: The name of the node.
+            overload: The FX overload of the node. It's None if the node is default overload.
             diagnostic_context: The diagnostic context to use for reporting errors.
 
         Returns:
             Set of function overloads.
         """
+
         function_group = None
+        custom_function_group = None
 
-        if self.onnx_registry.is_registered_op(aten_name, self.opset_version):
-            function_group = self.onnx_registry.get_function_group(aten_name)
+        if self.onnx_registry.is_registered_op(
+            domain=domain, op_name=op_name, overload=overload
+        ):
+            function_group = self.onnx_registry.get_functions(
+                domain=domain, op_name=op_name, overload=overload
+            )
+            custom_function_group = self.onnx_registry._get_custom_functions(
+                domain=domain, op_name=op_name, overload=overload
+            )
 
-        # Fall back to overloadpacket name: eg: aten.add.Tensor -> aten::add
+        # NOTE: Fall back to overloadpacket name: eg: aten.add.Tensor -> aten::add
+        # Support default fallback for overloadpacket.
         elif hasattr(
             node.target, "overloadpacket"
         ) and self.onnx_registry.is_registered_op(
-            node.target.overloadpacket._qualified_op_name,  # type: ignore[union-attr]
-            self.opset_version,
+            domain=domain, op_name=op_name, overload=None
         ):
-            function_group = self.onnx_registry.get_function_group(
-                node.target.overloadpacket._qualified_op_name  # type: ignore[union-attr]
+            function_group = self.onnx_registry.get_functions(
+                domain=domain, op_name=op_name, overload=None
+            )
+            custom_function_group = self.onnx_registry._get_custom_functions(
+                domain=domain, op_name=op_name, overload=None
             )
 
         if function_group is not None:
-            # TODO(titaiwang): dispatch opset version.
-            dispatched_version = _dispatch_opset_version(
-                self.opset_version, function_group.support_opset()
-            )
-            if dispatched_version is not None:
-                function_overloads = function_group.get(dispatched_version)
-                if function_overloads is not None:
-                    return function_overloads
+            return function_group, custom_function_group
```

Comment:
Thoughts on just returning a list/tuple to preserve ordering? Do we need membership checking here (with a set)?

Reply:
Could you share more of this idea? I don't follow.

Reply:
I will turn whole registry to dict[str: list[...]]

Reply:
Done